test/document_test.dart

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) 2011, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:markdown/markdown.dart';
+import 'package:test/test.dart';
+
+void main() {
+  group('Document', () {
+    test('guardHtml prevents less than and ampersand escaping', () {
+      var document = new Document(guardHtml: true);
+      var result = document.parseInline('< &');
+      expect(result, hasLength(1));
```

Comment:
I tried:
```dart
var expected = [new Text('< &')];
expect(result, equals(expected));
```
But I got:
```
  Expected: [Instance of 'Text']
    Actual: [Instance of 'Text']
     Which: was <Instance of 'Text'> instead of <Instance of 'Text'> at location [0]
```

Reply:
Ouch, sorry about that. Want to add a `toString()` to [Text](https://github.com/dart-lang/markdown/blob/master/lib/src/ast.dart#L59)?

Reply:
I tried adding this to `Text`:

```dart
String toString() => '<text: $text>';
```

Test:
```dart
test('guardHtml prevents less than and ampersand escaping', () {
  var document = new Document(guardHtml: true);
  var result = document.parseInline('< &');
  var expected = [new Text('< &')];
  expect(result, expected);
});
```

Result:
```
00:24 +1586 -1: test/document_test.dart: Document guardHtml prevents less than and ampersand escaping [E]                                                             
  Expected: [Text:<text: < &>]
    Actual: [Text:<text: < &>]
     Which: was Text:<<text: < &>> instead of Text:<<text: < &>> at location [0]
  
  package:test                  expect
  test/document_test.dart 14:7  main.<fn>.<fn>
  
00:29 +2484 -1: Some tests failed.  
```